source/vibe/db/postgresql/package.d

```diff
@@ -107,18 +108,19 @@ class __Conn : dpq2.Connection
     {
         import vibe.core.core;
 
-        auto sock = this.socket();
-
-        sock.blocking = false;
+        scope sock = this.socket();
+        auto triggerMode = FileDescriptorEvent.Trigger.any;
+        debug version(Have_vibe_core) triggerMode = FileDescriptorEvent.Trigger.read;
+        auto event = createFileDescriptorEvent(sock.handle, triggerMode);
         scope(exit)
         {
-            sock.blocking = true;
-            destroy(sock);
+            destroy(event); // Prevents 100% CPU usage
+            version(Have_vibe_core)
+                GC.clrAttr(cast(void*)sock, GC.BlkAttr.FINALIZE);
```

Comment:
Maybe will be better just duplicate Socket for vibe-core "destroyer counter" to avoid these dangerous manipulations?

Reply:
Brr, stop. socket_t duplication is fast operation, as I remember? Thus, we can duplicate sock.handle() result (by `core.sys.posix.unistd.dup`) for `createFileDescriptorEvent`

Reply:
This is possible, but I do not like it, i would prefer not tu use this.socket anyway, because main problem here is with std.socket.Socket, which is realy bad, because it is class and it is try to close socket in destructor whitch could throw an exception

Reply:
Here is already used duplicated Socket (scope sock = this.socket(); line). I am not propose to duplicate it too - just duplicate socket_t wrapped inside of sock

Reply:
> because it is class and it is try to close socket in destructor whitch could throw an exception

If duplicated socket will be closed duplicate will work ok.

Reply:
@denizzzka I understand, right know I am trying it :)